src/JsonSchema/Constraints/EnumConstraint.php

```diff
@@ -41,6 +41,31 @@ public function check($element, $schema = null, $path = null, $i = null)
             }
         }
 
-        $this->addError($path, "Does not have a value in the enumeration " . print_r($schema->enum, true), 'enum', array('enum' => $schema->enum,));
+        $this->addError($path, "value is not in enumeration: [" . $this->enumToString($schema->enum) . "]", 'enum', array('enum' => $schema->enum,));
+    }
+
+    /**
+     * @param array $enum
+     * @return string
+     */
+    private function enumToString(array $enum)
+    {
+        $enumString = '';
+
+        foreach ($enum as $value) {
+            if (is_array($value)) {
+                $enumString += '[' . $this->enumToString($value) . ']' . ', ';
+                continue;
+            }
+
+            if (is_object($value)) {
```

Comment:
@wpillar to be honest, I don't really like this approach. I feel like the code could be better, using arrays and `implode` instead of lots of less-than-efficient concatenations and that funny `rtrim` at the end. That being said, if you can rally a few approvals from the community, I won't hold it up.